precompile/fee_config_manager.go

```diff
@@ -0,0 +1,230 @@
+		MinBaseFee:               new(big.Int).SetBytes(returnPackedElement(input, 2)),
+		TargetGas:                new(big.Int).SetBytes(returnPackedElement(input, 3)),
+		BaseFeeChangeDenominator: new(big.Int).SetBytes(returnPackedElement(input, 4)),
+		MinBlockGasCost:          new(big.Int).SetBytes(returnPackedElement(input, 5)),
+		MaxBlockGasCost:          new(big.Int).SetBytes(returnPackedElement(input, 6)),
+		BlockGasCostStep:         new(big.Int).SetBytes(returnPackedElement(input, 7)),
+	}, nil
+}
+
+func GetFeeConfig(stateDB StateDB) (FeeConfig, error) {
+	if !stateDB.Exist(FeeConfigManagerAddress) {
+		return FeeConfig{}, nil
+	}
+	feeConfig := FeeConfig{}
+	for i := minKey; i <= maxKey; i++ {
+		val := stateDB.GetState(FeeConfigManagerAddress, common.Hash{byte(i)})
+		switch i {
+		case gasLimitKey:
+			feeConfig.GasLimit = new(big.Int).Set(val.Big())
+		case targetBlockRateKey:
+			feeConfig.TargetBlockRate = new(big.Int).Set(val.Big())
+		case minBaseFeeKey:
+			feeConfig.MinBaseFee = new(big.Int).Set(val.Big())
+		case targetGasKey:
+			feeConfig.TargetGas = new(big.Int).Set(val.Big())
+		case baseFeeChangeDenominatorKey:
+			feeConfig.BaseFeeChangeDenominator = new(big.Int).Set(val.Big())
+		case minBlockGasCostKey:
+			feeConfig.MinBlockGasCost = new(big.Int).Set(val.Big())
+		case maxBlockGasCostKey:
+			feeConfig.MaxBlockGasCost = new(big.Int).Set(val.Big())
+		case blockGasCostStepKey:
+			feeConfig.BlockGasCostStep = new(big.Int).Set(val.Big())
+		default:
+			return FeeConfig{}, fmt.Errorf("unknown field key %d", i)
+		}
+	}
+	return feeConfig, nil
+}
+
+func setFeeConfig(stateDB StateDB, feeConfig FeeConfig) error {
+	for i := minKey; i <= maxKey; i++ {
+		var hashInput common.Hash
+		switch i {
+		case gasLimitKey:
+			hashInput = common.BigToHash(feeConfig.GasLimit)
+		case targetBlockRateKey:
+			hashInput = common.BigToHash(feeConfig.TargetBlockRate)
+		case minBaseFeeKey:
+			hashInput = common.BigToHash(feeConfig.MinBaseFee)
+		case targetGasKey:
+			hashInput = common.BigToHash(feeConfig.TargetGas)
+		case baseFeeChangeDenominatorKey:
+			hashInput = common.BigToHash(feeConfig.BaseFeeChangeDenominator)
+		case minBlockGasCostKey:
+			hashInput = common.BigToHash(feeConfig.MinBlockGasCost)
+		case maxBlockGasCostKey:
+			hashInput = common.BigToHash(feeConfig.MaxBlockGasCost)
+		case blockGasCostStepKey:
+			hashInput = common.BigToHash(feeConfig.BlockGasCostStep)
+		default:
+			return fmt.Errorf("unknown field key %d", i)
+		}
+		stateDB.SetState(FeeConfigManagerAddress, common.Hash{byte(i)}, hashInput)
+	}
+	return nil
+}
+
+// createMintNativeCoin checks if the caller is permissioned for minting operation.
+// The execution function parses the [input] into native coin amount and receiver address.
+func createSetFeeConfig(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
```

Comment:
Can we change this function name to setFeeConfig? The reason it is create for the allow list functionality is that we are actually creating the function that will be used, but this function is just used directly.